build-test.cmd

```diff
@@ -6,8 +6,17 @@ set __BuildArch=x64
 set __VCBuildArch=x86_amd64
 set __BuildType=Debug
 set __BuildOS=Windows_NT
-set __VSVersion=vs2015
-set __VSToolsRoot=%VS140COMNTOOLS%
+
+:: Default to highest Visual Studio version available
+if defined VS150COMNTOOLS (
```

Comment:
Additionally, even if we have manual detection for the user passing in the tool version they wish to use, we should fail if the user is already in a Developer Command Prompt and they have specified a different version. This is because the environment variables and path settings can overlap and lead to weird behaviors.